addons/mail/static/src/models/composer/composer.js

```diff
@@ -656,7 +656,7 @@ function factory(dependencies) {
             const mentions = [];
             for (const partner of this.mentionedPartners) {
                 const placeholder = `@-mention-partner-${partner.id}`;
-                const text = `@${owl.utils.escape(partner.name)}`;
+                const text = _.escape(`@${owl.utils.escape(partner.name)}`);
```

Comment:
@ged-odoo which option is better:

* call ``_.escape`` after owl.utils.escape (current version)
* don't call ``_.escape`` in [escapeAndCompactTextContent](https://github.com/odoo/odoo/blob/329b00365e7b7702eb4923443c99ddac13340ae8/addons/mail/static/src/js/utils.js#L147-L156)
* implement ``_.escape`` in ow.utils.escape

?

Reply:
What is the current difference between `_.escape` and `ow.utils.escape`?

Shouldn't the fix be to use `ow.utils.escape` instead of `_.escape` inside `escapeAndCompactTextContent`, if that is what the result of the current line is compared against?

Reply:
`_.escape("putin's spokesman") ->"putin&#x27;s spokesman"`

Maybe this `&#x27` is better for mail body?
If not, I'm ok to use solution n.2

Reply:
I don't even remember why we escape in this flux in the first place, since the `body` in this method is sent to the server, which is already taking care of sanitizing.

Also escaping `'` and `"` seems useless inside `body` as those characters only allow injection inside attributes, and we never display body inside an attribute. And as far as I know we don't even have a `t-raw` equivalent for attributes anyway, `qweb` appears to be always escaping attributes by default. Hopefully I'm correct on this and that would be why `escape` from `owl` doesn't bother replacing those.

So according to this

- Points 1. and 3. seems overkill
- Point 2. seems safe + good
- completely removing escape from here and from escapeAndCompactTextContent might be ok too (but is less safe, would need double check)

Reply:
@nim-odoo what do you think?

Reply:
I'd also go for 2.